src/cryptsetup/cryptsetup.c

```diff
@@ -179,6 +180,13 @@ static int parse_one_option(const char *option) {
         } else if (streq(option, "tcrypt-system")) {
                 arg_type = CRYPT_TCRYPT;
                 arg_tcrypt_system = true;
+        } else if (streq(option, "tcrypt-veracrypt")) {
+#ifdef CRYPT_TCRYPT_VERA_MODES
+                arg_type = CRYPT_TCRYPT;
+                arg_tcrypt_veracrypt = true;
+#else
+                log_error("This version of cryptsetup does not support VeraCrypt; ignoring tcrypt-veracrypt.");
```

Comment:
Hm, I'm not convinced that this fallback is OK.
The question is, what should happen if "tcrypt-veracrypt" is added, but we don't support it.
Should it:
1. fall back to "tcrypt",
2. error out,
3. just warn (as this patch does)?
If fall back to "plain" tcrypt makes sense, then 1. would be best, otherwise 2. Option 3. seems least desirable.